app/controllers/submissions_controller.rb

```diff
@@ -203,25 +203,36 @@ def collect_submissions
     end
     assignment = Assignment.includes(:groupings).find(params[:assignment_id])
     groupings = assignment.groupings.find(params[:groupings])
-    partition = groupings.partition do |grouping|
+    collectable = []
+    some_before_due = false
+    some_released = Grouping.joins(current_submission_used: :results)
+                            .where('results.released_to_students': true)
+                            .where(id: groupings)
```

Comment:
I'd add a `.pluck(:id).to_set` here. It's not a big deal in most cases, but would matter somewhat if the instructor accidentally selects a whole bunch of released submissions. (I just tried it one ~200 submissions from our seed data.)